sdk/dotnet/Pulumi/Deployment/Deployment.Runner.cs

```diff
@@ -29,6 +30,9 @@ private class Runner : IRunner
             /// exiting once the set becomes empty.
             /// </summary>
             private readonly Dictionary<Task, List<string>> _inFlightTasks = new Dictionary<Task, List<string>>();
+            private readonly List<Exception> _exceptions = new List<Exception>();
```

Comment:
Hrm. Is this useful? This made me wonder about why we have this entire class in here, and it sounds like at a very high level it's there because "new Resource()" completes in a fire-and-forget manner and the background exceptions need to go somewhere, so presumably they go here? Is lifecycle of Runner pretty contained so we won't have this list growing indefinitely in a server or smth?

Reply:
Lifecycle of the `IRunner` is tied to the lifecycle of the `Deployment` [here](https://github.com/pulumi/pulumi/blob/98211718ad2b4466c2efc3b452a4c52b0e468872/sdk/dotnet/Pulumi/Deployment/Deployment_Run.cs#L190), so only for the duration of the linked method and then it will no longer be referenced. So the exception list should _at most_ be the exceptions only from a single deployment before the `IRunner` is de-referenced and hit by GC